repositories/productRepository.go

```diff
@@ -236,9 +239,12 @@ func (repository *productRepository) UpdateProduct(product models.Product) error
 		log.Println("An error has occured during marshaling the product, ", err)
 	}
 
-	repository.kafka.WriteMessage(string(jsonMessage), product.ID)
-
-	return tx.Commit(context.Background())
+	err = repository.kafka.WriteMessage(string(jsonMessage), product.ID)
+	if err != nil {
+		return err
+	} else {
```

Comment:
A bit nitpicky, but else is redundant here. Same for creation and deletion.

Reply:
I'll approve the PR once you fix this small issue. Don't have anything else to add.

Reply:
Fixed it, commit is now after if statement